### ISSUE TITLE
TF2 tf.distribute.MirroredStrategy() failed

### ISSUE THREAD

**Issue by xiaohu2015** [tensorflow/tensorflow#28101]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):  try the code https://github.com/tensorflow/docs/blob/master/site/en/r2/tutorials/distribute/keras.ipynb 
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2.0 alpha
- Python version: 3.6
- Bazel version (if compiling from source): 0.23
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 9.0 / 7.0
- GPU model and memory: P40

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
File "keras_mnist.py", line 79, in <module>
    model.fit(train_dataset, epochs=10, callbacks=callbacks)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/engine/training.py", line 791, in fit
    initial_epoch=initial_epoch)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/engine/training.py", line 1515, in fit_generator
    steps_name='steps_per_epoch')
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/engine/training_generator.py", line 257, in model_iteration
    batch_outs = batch_function(*batch_data)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/engine/training.py", line 1258, in train_on_batch
    self._make_fit_function()
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/engine/training.py", line 2177, in _make_fit_function
    '_fit_function', [self.total_loss] + metrics_tensors)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/engine/training.py", line 2149, in _make_train_function_helper
    params=self._collected_trainable_weights, loss=self.total_loss)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 454, in get_updates
    return [self.apply_gradients(grads_and_vars)]
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 406, in apply_gradients
    self._distributed_apply, args=(grads_and_vars,), kwargs={"name": name})
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/distribute/distribute_lib.py", line 1375, in merge_call
    return self._merge_call(merge_fn, args, kwargs)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/distribute/distribute_lib.py", line 1382, in _merge_call
    return merge_fn(self._strategy, *args, **kwargs)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/keras/optimizer_v2/optimizer_v2.py", line 440, in _distributed_apply
    var, apply_grad_to_update_var, args=(grad,), group=False))
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/distribute/distribute_lib.py", line 1175, in update
    return self._update(var, fn, args, kwargs, group)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/distribute/distribute_lib.py", line 1546, in _update
    return self._update_non_slot(var, fn, (var,) + tuple(args), kwargs, group)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/distribute/distribute_lib.py", line 1551, in _update_non_slot
    with ops.colocate_with(colocate_with), UpdateContext(colocate_with):
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 4261, in _colocate_with_for_gradient
    with self.colocate_with(op, ignore_existing):
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 4310, in colocate_with
    op = _op_to_colocate_with(op)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 6556, in _op_to_colocate_with
    return internal_convert_to_tensor_or_indexed_slices(v, as_ref=True).op
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1347, in internal_convert_to_tensor_or_indexed_slices
    value, dtype=dtype, name=name, as_ref=as_ref)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1186, in internal_convert_to_tensor
    ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
  File "/data2/huye/anaconda3/envs/tf2/lib/python3.6/site-packages/tensorflow/python/distribute/values.py", line 731, in _tensor_conversion_mirrored
    assert not as_ref
AssertionError

**Describe the expected behavior**

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by achandraa** [comment]
I tried reproducing the issue through colab [link](https://github.com/tensorflow/docs/blob/master/site/en/r2/tutorials/distribute/keras.ipynb) provided but the code executed without any error. Can you try once again and let us know if that still gives error. Thanks!

**Comment by achandraa** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
